test/junit/scala/collection/immutable/ListSetTest.scala

```diff
@@ -50,4 +52,32 @@ class ListSetTest {
     val s = ListSet(1, 2, 3, 5, 4)
     assertEquals(List(1, 2, 3, 5, 4), s.iterator.toList)
   }
+
+  @Test
+  def hasCorrectOrderAfterPlusPlus(): Unit = {
```

Comment:
It's probably worth adding a couple more interesting test cases like the ones in this comment: https://github.com/scala/bug/issues/12316#issuecomment-761888794

Arguably this test suite should be using property based testing (e.g. generate two Sets, convert to Lists, convert to ListSets, concatenate the ListSets and verify the result matches the concatenated Lists), but I'm not at all familiar with testing conventions or other considerations in this project.

Reply:
feel free to add some tests